src/shared/telemetry/telemetryUtils.ts

```diff
@@ -42,15 +57,16 @@ export function registerCommand<T>({
                 throw e
             } finally {
                 const endTime = new Date()
-                const datum = result && result.datum ? result.datum : defaultMetricDatum(command)
+                const datum = result && result.datum ?
+                    result.datum : defaultMetricDatum(telemetryName ? telemetryName.name : command)
                 if (!datum.metadata) {
                     datum.metadata = new Map()
                 }
-                datum.metadata.set('hasException', `${hasException}`)
+                datum.metadata.set('result', hasException ? 'Failed' : 'Succeeded')
```

Comment:
Is there a test for this and if not is it possible for there to be?

Reply:
✔️